dbms/src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -54,6 +55,9 @@ PushingToViewsBlockOutputStream::PushingToViewsBlockOutputStream(
         output = storage->write(query_ptr, context.getSettingsRef());
         replicated_output = dynamic_cast<ReplicatedMergeTreeBlockOutputStream *>(output.get());
     }
+
+    threads.reserve(views.size());
+    exceptions.resize(views.size());
```

Comment:
If you use `ThreadPool`, exceptions will be rethrown automatically.

Reply:
Makes sense, I'll rework it with ThreadPool, thanks!